collector/processor/batchprocessor/batch_processor.go

```diff
@@ -254,7 +326,35 @@ func (b *shard) resetTimer() {
 
 func (b *shard) sendItems(trigger trigger) {
 	sent, bytes, err := b.batch.export(b.exportCtx, b.processor.sendBatchMaxSize, b.processor.telemetry.detailed)
-	if err != nil {
+
+	numItemsBefore := b.totalSent
+	numItemsAfter := b.totalSent + sent
+	// The current batch can contain items from several different producers. Ensure each producer gets a response back.
```

Comment:
Note: in a follow-on PR I'd like the `export` to happen in the background -- it's going to mean running through this loop before calling export, so we know the set of complete/pending responses that will be returned when the response comes back.